dali/base/danqs.cpp

```diff
@@ -556,6 +560,8 @@ class CDaliNamedQueueServer: public IDaliServer, public Thread, implements IConn
 
     void processMessage(CMessageBuffer &mb)
     {
+        hpccMetrics::ScopedGaugeUpdater gaugeUpdater(*pNqActiveRequests, 1);
```

Comment:
This isn't very useful.  Its value will be either 0 or 1, indicating whether dali is currently processing a message.

Reply:
A general efficiency concern - this involves two atomic operations.  If you recorded the "dali.nq_requests.processed" instead it would only involve 1 atomic operation.  You could publish dali.nq_requests.active by subtracting them.  I suspect it is more useful to publish a counter of the number processed.
@rpastrana how easy is it in prometheus to subtract two counters and display the result?

Reply:
A different efficiency concern is that this code is single-threaded so none of the metric increments need to be atomic.  There should be a version of the increment that avoids the atomic penalty.  (approx 10ns on intel for reference, different on ARM/POWER8)